monai/transforms/croppad/dictionary.py

```diff
@@ -377,6 +378,59 @@ def __call__(self, data: Mapping[Hashable, np.ndarray]) -> Dict[Hashable, np.nda
         return d
 
 
+class RandWeightedCropd(Randomizable, MapTransform):
+    """
+    Samples a list of `num_samples` image patches according to the provided `weight_map`.
+
+    Args:
+        keys: keys of the corresponding items to be transformed.
+            See also: :py:class:`monai.transforms.compose.MapTransform`
+        w_key: key for the weight map. The corresponding value will be used as the sampling weights,
+            it should be a single-channel array in size, for example, `(1, spatial_dim_0, spatial_dim_1, ...)`
+        spatial_size: the spatial size of the image patch e.g. [224, 224, 128].
+            If its components have non-positive values, the corresponding size of `img` will be used.
+        num_samples: number of samples (image patches) to take in the returned list.
+
+    See Also:
+        :py:class:`monai.transforms.RandWeightedCrop`
+    """
+
+    def __init__(self, keys: KeysCollection, w_key: str, spatial_size: Union[Sequence[int], int], num_samples: int = 1):
+        super().__init__(keys)
+        self.spatial_size = ensure_tuple(spatial_size)
+        self.w_key = w_key
+        self.num_samples = int(num_samples)
+        self.centers: List[np.ndarray] = []
+
+    def randomize(self, weight_map: np.ndarray) -> None:
+        self.centers = weighted_patch_samples(
+            spatial_size=self.spatial_size, w=weight_map[0], n_samples=self.num_samples, r_state=self.R
+        )
+
+    def __call__(self, data: Mapping[Hashable, np.ndarray]) -> List[Dict[Hashable, np.ndarray]]:
+        d = dict(data)
+        self.randomize(d[self.w_key])
+        _spatial_size = fall_back_tuple(self.spatial_size, d[self.w_key].shape[1:])
+
+        results: List[Dict[Hashable, np.ndarray]] = [dict() for _ in range(self.num_samples)]
+        for key in data.keys():
+            if key in self.keys:
+                img = d[key]
+                if img.shape[1:] != d[self.w_key].shape[1:]:
+                    raise ValueError(
+                        f"data {key} and weight map {self.w_key} spatial shape mismatch: "
+                        f"{img.shape[1:]} vs {d[self.w_key].shape[1:]}."
+                    )
+                for i, center in enumerate(self.centers):
+                    cropper = SpatialCrop(roi_center=center, roi_size=_spatial_size)
+                    results[i][key] = cropper(img)
+            else:
+                for i in range(self.num_samples):
+                    results[i][key] = data[key]
+
+        return results
```

Comment:
If `num_samples` is 1 it might be easier to return `results[0]` here so that further transforms expecting a dictionary can function with this one.

Reply:
this is to be consistent with the other few transforms. if we want to change this behaviour we should address it with a separate ticket.